### PR TITLE
Add external dependencies check

### PR DESCRIPTION
**Pull request by hongyh13**
Check external dependencies before running job hosting, wait until all the external dependencies are ready.

### CODE REVIEW

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.DataWriter/Azure/AzureBlobContainerClient.cs [hunk 1]
```diff
@@ -428,5 +445,10 @@ public async IAsyncEnumerable<PathItem> ListPathsAsync(string directory, [Enumer
             }
             while (item != null);
         }
+
+        private static bool IsAuthenticationError(RequestFailedException exception) =>
+            string.Equals(exception.ErrorCode, AzureStorageErrorCode.NoAuthenticationInformationErrorCode, StringComparison.OrdinalIgnoreCase) ||
```
Comment (moria97) [inline FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.DataWriter/Azure/AzureBlobContainerClient.cs:450]: Shall we check only 403 code from [doc](https://learn.microsoft.com/en-us/rest/api/storageservices/common-rest-api-error-codes)

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.DataWriter/Azure/AzureBlobContainerClient.cs [hunk 2]
```diff
@@ -108,20 +111,29 @@ private BlobContainerClient BlobContainerClient
...
-                                var tempBlobContainerClient = new BlobContainerClient(
-                                    _storageUri,
-                                    externalTokenCredential);
-
-                                InitializeBlobContainerClient(tempBlobContainerClient);
-                                _blobContainerClient = tempBlobContainerClient;
-                            }
-                            catch (Exception ex)
+                            bool retry = true;
+                            while (retry)
```
Comment (QuanWanxx) [inline FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.DataWriter/Azure/AzureBlobContainerClient.cs]: Can we leverage the [Polly](https://github.com/App-vNext/Polly) to retry?

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.DataWriter/Azure/AzureBlobContainerClient.cs [hunk 3]
```diff
@@ -155,19 +164,27 @@ private DataLakeFileSystemClient DataLakeFileSystemClient
...
             catch (Exception ex)
             {
                 _diagnosticLogger.LogError($"Create container {blobContainerClient.Name} failed. Reason: {ex.Message}");
-                _logger.LogInformation(ex, $"Create container {blobContainerClient.Name} failed. Reason: {ex.Message}");
-                throw new AzureBlobOperationFailedException($"Create container {blobContainerClient.Name} failed.", ex);
+                _logger.LogError(ex, $"Failed to create container {blobContainerClient.Name}.");
+                throw new AzureBlobOperationFailedException($"Failed to create container {blobContainerClient.Name}.", ex);
             }
+
+            return false;
```
Comment (QuanWanxx) [inline FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.DataWriter/Azure/AzureBlobContainerClient.cs]: Feel a little strange for returning false if success, maybe return true and change outer logic?

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.DataWriter/Azure/AzureBlobContainerClient.cs (hunk 2 above)
Comment (QuanWanxx) [reply FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.DataWriter/Azure/AzureBlobContainerClient.cs]: Ignore

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.DataWriter/Azure/AzureBlobContainerClient.cs (hunk 3 above)
Comment (QuanWanxx) [reply FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.DataWriter/Azure/AzureBlobContainerClient.cs]: Ignore

**Review by moria97 — APPROVED** [review]

**Pull request merged**